sdk/test/opentelemetry/sdk/trace/export/batch_span_processor_test.rb

```diff
@@ -37,6 +37,12 @@ def export(batch)
     def shutdown; end
   end
 
+  class TestTimeoutExporter < TestExporter
+    def export(batch)
+      raise Timeout::Error
```

Comment:
This isn't really testing the functionality (that the processor enforces the timeout). I'd rather see a test that sets the timeout quite low, triggers the export and ensures that a) the exporter.export was called and b) it was interrupted by a timeout.

Reply:
I've rewritten the test, demonstrating that the exporter was `:called` and does not get to the `:not_interrupted` state.

Reply:
:heavy_check_mark: